### PR TITLE
[FEATURE] Allow writing and reading a sketch file

### PR DESCRIPTION
**Pull request by smehringer**
Summary
---------------------

This PR changes the following:

- [x] It removes the anyway unused function `read_hll_files_into`
- [x] It changes the semantic of the cmd option `--output-sketches-to` *1
- [x] It add to the semantic of the cmd option `--input` *2
- [x] Adds/changes test for the new behavior

*1 `--output-sketches-to`
---------------------

Before this PR `--output-sketches-to DIR` would write one sketch file for each input file.
This had several disadvantages: many files, not scalable to millions of user bins, with additional minhashes there are even more files, no direct information about the input filenames as well as the meta information of how the sketches were generated (k,w,sketch-bits).

Instead, now the option `--output-sketches-to FILENAME`, writes a sketch file (`chopper::sketch::sketch_file`) that contains the chopper config, filenames and the sketches, the latter two in the same order. This way, only a single file is written will all important information contained. Slight disadvantage: You can (currently) not read a fraction of the input, e.g. "just the first 1000 sketches of the data set". You would need to create a separate sketch file for this.

*2 `--input`
------------

Before this PR, only a list of filenames would be allowed as input to choppper (layout).
After this PR, a sketch file is also a valid source of input as it contains the filenames and the sketches.

### CODE REVIEW

**Comment by seqan-actions** [comment]
Documentation preview available at https://docs.seqan.de/preview/seqan/chopper/254

File: include/chopper/sketch/sketch_file.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        archive(CEREAL_NVP(version));
+
+        archive(CEREAL_NVP(chopper_config));
+        archive(CEREAL_NVP(filenames));
+        archive(CEREAL_NVP(hll_sketches));
+        archive(CEREAL_NVP(minHash_sketches));
+    }
+};
+
+} // namespace chopper::sketch
```
Comment (smehringer) [inline include/chopper/sketch/sketch_file.hpp:45]:
```suggestion
} // namespace chopper::sketch

```

**Review by eseiler — APPROVED** [review]
I pushed a review commit. If the changes are fine, you can squash and merge :)

**Comment by eseiler** [comment]
One more thing:
Do you use the stuff in https://github.com/seqan/chopper/blob/main/src%2Fsketch%2Foutput.cpp ?
If not, it could be removed.

**Pull request merged**